src/main/java/com/mathworks/ci/MatlabInstallationAxis.java

```diff
@@ -0,0 +1,71 @@
+package com.mathworks.ci;
+
+/**
+ * Copyright 2020 The MathWorks, Inc.
+ *
+ * Describable class for MATLAB Axis that provides a list of configured MATLAB installation for
+ * generating matrix configurations.
+ *
+ */
+
+import hudson.Extension;
+import hudson.matrix.Axis;
+import hudson.matrix.AxisDescriptor;
+import hudson.matrix.MatrixProject;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class MatlabInstallationAxis extends Axis {
+
+    @DataBoundConstructor
+    public MatlabInstallationAxis(List<String> values) {
+        super(Message.getValue("Axis.matlab.key"), evaluateValues(values));
+    }
+
+    static private List<String> evaluateValues(List<String> values) {
+        // Add default configuration is values are null or not selected.
+        if (values == null || values.isEmpty()) {
+            values = new ArrayList<>(Arrays.asList("default"));
+        }
+        return values;
+    }
+
+    @Extension
+    public static class DescriptorImpl extends AxisDescriptor{
+        private final String useMatlabWarning = Message.getValue("Axis.use.matlab.warning");
```

Comment:
would separate final string make any difference if the message is pulled from a static property file.

Reply:
Not on performance but it does add to readability and emphasize that it's a constant value in classes context.

https://softwareengineering.stackexchange.com/questions/250804/why-to-declare-a-string-as-final-and-then-use-it

Reply:
I guess again the benefits will boil down to reduce the total line number in the code. I would choose one over the other just to reduce the number of extra lines of code

Reply:
Done.